op-e2e/setup.go

```diff
@@ -465,6 +465,8 @@ func (cfg SystemConfig) Start(_opts ...SystemConfigOption) (*System, error) {
 			if p, ok := p2pNodes[name]; ok {
 				return p, nil
 			}
+			// TODO: This will need to be modified to use sys.Mocknet.AddPeerWithPeerstore so we can make the peerstore
+			// an extended peerstore.
 			h, err := sys.Mocknet.GenPeer()
```

Comment:
This was an unfortunate discovery.  Switching to specifying the peerstore ourselves will require duplicating a bit of code that `GenPeer` previously did.  Alternatively we could introduce the `ExtendedPeerStore` wrapper later rather than it being the actual host `Peerstore` but it feels a lot nicer to have everything go through the wrapper.

Reply:
I updated the libp2p PR to so that GenPeer can take a PeerStore and/or ConnectionManager so we can duplicate code in the short term but hopefully migrate to those new APIs in a future go-libp2p version.